src/Akka.Persistence.Sql/Query/QueryThrottler.cs

```diff
@@ -40,31 +59,72 @@ internal sealed class ReturnQueryStart
     private ReturnQueryStart() { }
 }
 
+#region Test classes
+
+/// <summary>
+/// For testing purposes
+/// </summary>
+internal sealed class GetUsedPermits
+{
+    public static readonly GetUsedPermits Instance = new();
+    private GetUsedPermits() { }
+}
+
+/// <summary>
+/// For testing purposes
+/// </summary>
+internal sealed class GetPendingRequests
+{
+    public static readonly GetPendingRequests Instance = new();
+    private GetPendingRequests() { }
+}
+
+internal sealed class GetWatchCount
+{
+    public static readonly GetWatchCount Instance = new();
+    private GetWatchCount() { }
+}
+
+#endregion
+
 /// <summary>
 /// Token bucket throttler that grants queries permissions to run each iteration
 /// </summary>
 /// <remarks>
-/// Works identically to the RecoveryPermitter built into Akka.Persistence.
+/// Works almost identically to the RecoveryPermitter built into Akka.Persistence.
+/// 
+/// NOTE: Since this permitter works with Ask and we can't death watch an Ask temp
+///       actor, there is a possibility that a permit requester failed to return
+///       the request permit.
+/// 
+///       ALWAYS USE A TRY...FINALLY BLOCK WHEN USING ASK AND RETURN THE PERMIT IN
+///       THE FINALLY BLOCK
 /// </remarks>
 internal sealed class QueryThrottler : ReceiveActor
 {
-    private readonly LinkedList<IActorRef> _pending = new();
+    private readonly LinkedList<PendingRequest> _pending = new();
```

Comment:
.... The use of LinkedList here is curious, if nothing else I wonder if an UnrolledLinkedList or similar would be worth considering (I'm guessing we use this for performance?)

Reply:
Its mostly because the .NET linked list is a doubly linked list implementation, the read/write time is constant for the first and last element (O(1)). It is just the perfect fit for the implementation.